lib/util/propTypes.js

```diff
@@ -970,16 +970,6 @@ module.exports = function propTypesInstructions(context, components, utils) {
           }
         });
       } else {
-        // check if its a valid generic type when `X<{...}>`
-        if (
-          param.typeAnnotation
-          && param.typeAnnotation.typeAnnotation
-          && param.typeAnnotation.typeAnnotation.type === 'TSTypeReference'
-          && param.typeAnnotation.typeAnnotation.typeParameters != null
-          && !isValidReactGenericTypeAnnotation(param.typeAnnotation.typeAnnotation)
-        ) {
-          return;
-        }
```

Comment:
i'm confused why this was here at all if removing it didn't cause a test to fail :-/

Reply:
Yes, I was wondering the same. I put it there because I thought we were only handling generic types that came from react for components declared as `const Component = (props: IProps) => {...}`. This broke the earlier behavior, my fault - will be careful from next time.